src/Utility/FunctionInvocationPerformanceStopwatch.cs

```diff
@@ -0,0 +1,44 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Utility
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+
+    internal class FunctionInvocationPerformanceStopwatch
+    {
+        public enum Checkpoint
+        {
+            DependenciesAvailable,
```

Comment:
Would it make sense to add a checkpoint for when the function app dependencies are being installed? This would be for the upgrade scenario.

Reply:
The main purpose of this PR is to measure cold start impact. By design, _upgrading_ dependencies is _not_ on the function invocation path, as it is performed on the background, so we would not encounter this checkpoint anywhere on this invocation chain.

If we are interested in measuring how long background upgrades take, we can already do it based on existing traces ("Installing function app required modules" etc.). For the background upgrade, precision is not that important, so we can rely on the regular log timestamps. I'm doing something special here because we want _at least_ 50-100 ms precision.

Reply:
Good point. Thanks for clarifying.